backend/src/pong/matchmaking-request/matchmaking-request.spec.ts

```diff
@@ -1,57 +1,137 @@
 			imports: [
 				ConfigModule.forRoot({ isGlobal: true }),
 				TypeOrmModule.forRootAsync({ useClass: TypeOrmConfigService }),
-				TypeOrmModule.forFeature([MatchmakingRequest]),
-				UserModule,
+				TypeOrmModule.forFeature([MatchmakingRequest, User, Game, GameInvite]),
 				AuthModule,
-				SharedModule,
-				AnimalModule,
 				ChatModule,
+				GameInvitesModule,
+				GameModule,
+				MatchmakingRequestModule,
 				MessageModule,
-				UserModule,
 				RoleModule,
-				GameModule,
+				SharedModule, // import for the JwtService
 				UserChatModule,
-				GameInvitesModule,
-				MatchmakingRequestModule,
+				UserModule,
 			],
-			providers: [MatchmakingRequestService],
-			controllers: [MatchmakingRequestController],
+			providers: [MatchmakingRequestService, UserService, JwtService],
+			controllers: [MatchmakingRequestController, UserController],
 		}).compile();
 
-		service = module.get<MatchmakingRequestService>(MatchmakingRequestService);
+		mmrService = module.get<MatchmakingRequestService>(
+			MatchmakingRequestService,
+		);
 		controller = module.get<MatchmakingRequestController>(
 			MatchmakingRequestController,
 		);
+		userController = module.get<UserController>(UserController);
+		userService = module.get<UserService>(UserService);
+
+		await seed();
+	});
+
+	afterAll(async () => {
+		const allRequests: MatchmakingRequest[] = await mmrService.findAll({
+			relations: { user: true },
+		});
+
+		for (let i = 0; i < allRequests.length; i++) {
+			await mmrService.remove(allRequests[i].id);
+		}
+		for (let i = 0; i < seededUsers.length; ++i) {
+			await userService.remove(seededUsers[i].id);
+		}
 	});
 
 	it('should be defined', () => {
-		expect(service).toBeDefined();
+		expect(mmrService).toBeDefined();
 		expect(controller).toBeDefined();
+		expect(userController).toBeDefined();
+	});
+
+	describe('findAll', () => {
+		it('should have been created with a relation to user', async () => {
+			const matchRequests: MatchmakingRequest[] = await mmrService.findAll({
+				relations: { user: true },
+			});
+			expect(matchRequests).toEqual(
+				expect.arrayContaining([
+					expect.objectContaining({
+						id: expect.any(Number),
+						created_at: expect.any(Date),
+						updated_at: expect.any(Date),
+						user: expect.any(User),
+					}),
+				]),
+			);
+		});
+	});
+
+	describe('create', () => {
+		it('should throw when trying to join with user that does not exist', async () => {
+			const nonExistantUserId: number = 9999;
+			await expect(
+				mmrService.save({ user: nonExistantUserId }),
+			).rejects.toThrow('FOREIGN KEY constraint failed');
+		});
+
+		it('should throw when trying to join with user that already has a gameRequest', async () => {
+			const nonExistantUserId: number = 1;
+			await expect(
+				mmrService.save({ user: nonExistantUserId }),
+			).rejects.toThrow(
+				'UNIQUE constraint failed: matchmaking_requests.invite_users',
+			);
+		});
```

Comment:
I don't think you catch these in the controller, so technically the raw database error will be shown to the user?
We probably need a try/catch in the controller to throw either not-found or some other relevant error

Reply:
Added `try{}catch` blocks